sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/PythonExternalTransform.java

```diff
@@ -179,16 +204,20 @@ Row buildOrGetKwargsRow() {
   // Types that are not one of following are considered custom types.
   // * Java primitives
   // * Type String
+  // * Type PythonCallableSource
+  // * Any Type explicitly annotated by withTypeHint()
   // * Type Row
-  private static boolean isCustomType(java.lang.Class<?> type) {
+  private boolean isCustomType(java.lang.Class<?> type) {
     boolean val =
         !(ClassUtils.isPrimitiveOrWrapper(type)
             || type == String.class
+            || type == PythonCallableSource.class
```

Comment:
Can this be generalized to all logical types somehow (instead of special casing PythonCallableSource) ?

Reply:
I think it's not possible. Currently, there's no way to automatically map a language type to a representation type in Java SDK. We need additional implementation for it.

Reply:
I would at least move this to a "isLogicalSchemaType()" method.